### PR TITLE
Move delay function so it is called per round instead of per transaction

### PR DESCRIPTION
**Pull request by JeffreyDallas**
The fix is to resolve regression test failed due to
exception from StoppableThreadImpl

https://github.com/swirlds/swirlds-platform-regression/issues/3202

Since a round may have many transactions, if calling delay() function many times within 
this round, the total time added inside **handleTransaction() ** may be beyond
5 second threshold and causing exception from **StoppableThreadImpl**

The solution is to move delay() function, so it is called per round instead of per transactions

```
2023-02-20 17:53:58.550 6452     ERROR EXCEPTION        <<platform-core: syncCaller-0 2>> StoppableThreadImpl: stack trace of thread-cons:
java.lang.Thread.sleep(Thread.java:-2)
	at com.swirlds.demo.platform.SyntheticBottleneckConfig.throttleIfNeeded(SyntheticBottleneckConfig.java:153)
	at com.swirlds.demo.platform.PlatformTestingToolState.delay(PlatformTestingToolState.java:703)
	at com.swirlds.demo.platform.PlatformTestingToolState.handleTransaction(PlatformTestingToolState.java:1110)
	at com.swirlds.demo.platform.PlatformTestingToolState.handleConsensusTransaction(PlatformTestingToolState.java:1042)
	at com.swirlds.demo.platform.PlatformTestingToolState.lambda$handleConsensusRound$8(PlatformTestingToolState.java:1034)
	at com.swirlds.demo.platform.PlatformTestingToolState$$Lambda$756/0x0000000801106578.accept(null:-1)
	at com.swirlds.common.system.Round.forEachEventTransaction(Round.java:95)
	at com.swirlds.demo.platform.PlatformTestingToolState.handleConsensusRound(PlatformTestingToolState.java:1032)
	at com.swirlds.platform.state.TransactionHandler.handleRound(TransactionHandler.java:180)
	at com.swirlds.platform.state.SwirldStateManagerDouble.handleConsensusRound(SwirldStateManagerDouble.java:154)
	at com.swirlds.platform.eventhandling.ConsensusRoundHandler.applyConsensusRoundToState(ConsensusRoundHandler.java:333)
	at com.swirlds.platform.eventhandling.ConsensusRoundHandler$$Lambda$558/0x0000000801006818.accept(null:-1)
	at com.swirlds.common.threading.framework.internal.QueueThreadImpl.doWork(QueueThreadImpl.java:234)
	at com.swirlds.common.threading.framework.internal.QueueThreadImpl$$Lambda$493/0x0000000800fa87d0.run(null:-1)
	at com.swirlds.common.threading.framework.internal.StoppableThreadImpl.doWork(StoppableThreadImpl.java:611)
	at com.swirlds.common.threading.framework.internal.StoppableThreadImpl.run(StoppableThreadImpl.java:217)
	at com.swirlds.common.threading.framework.internal.StoppableThreadImpl$$Lambda$495/0x0000000800fa93c0.run(null:-1)
	at java.lang.Thread.run(Thread.java:833)

```

Test result

https://swirldslabs.slack.com/archives/C03G7CBJJ06/p1677522668784719

### CODE REVIEW

**Review by jeromy-cannon — APPROVED** [review]

**Review by qnswirlds — DISMISSED** [review]
LGTM. But we better have an approval from the platform team as well.

**Review by jeromy-cannon — APPROVED** [review]

**Comment by cody-littley** [comment]
LGTM as long as this doesn't cause tests to fail

**Review by cody-littley — APPROVED** [review]

**Pull request merged**
